### ISSUE TITLE
need to add SparseEmbedding layer support

### ISSUE THREAD

**Issue by cmxcn** [tensorflow/tensorflow#41588]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Hi, can we add a SparseEmbedding layer into tf.keras.layers or make tf.keras.layers.Embedding support sparse embedding? 
The main idea is that in many cases like recommendation or CTR prediction models, there are a huge number of items to be embeded, but in each batch we only use a very small portion of it. Therefore, the SparseEmbedding layer should bring just the parameters that are actually used in this batch into the model parameters to be optimized. This is will provide a huge performance improvement, especially in Multiworker Distributed training as the network cost will be significantly decreased.
BTW, pytorch already have something similar like torch.nn.Embedding(sparse=True) https://pytorch.org/docs/master/generated/torch.nn.Embedding.html
**Will this change the current api? How?**
small refactor only. By adding a new Class or adding a construction parameter
**Who will benefit with this feature?**
in many cases like recommendation or CTR prediction models, there are a huge number of items to be embeded, but in each batch we only use a very small portion of it.
**Any Other info.**

**Comment by cmxcn** [comment]
Hi, may I know if there are any thoughts regarding this?

**Comment by cmxcn** [comment]
My bad.. it looks like tf.gather already provides sparse gradient, so no bother needed.

**Comment by tanzhenyu** [comment]
So it's not a bad idea to have Embedding layer to support sparse inputs, I can help on this if you need. Though you're right if you use simple gather ops it will do the same thing
